### PR TITLE
[SymbolGraphGen] add new flag to skip "protocol implementation" symbols

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
Resolves rdar://59899968

This PR introduces a new flag to affect SymbolGraphGen, available on the frontend and on `swift-symbolgraph-extract`: `-skip-protocol-implementations`. When present, this flag causes SymbolGraphGen to drop symbols that have an "inherited symbol", i.e. that are implementations of protocol requirements or are synthesized copies of protocol extensions. The conformance relationship between the type and the protocol itself remains, so that these relationships can be reconstructed.

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
@swift-ci Please smoke test

**Comment by QuietMisdreavus** [comment]
@swift-ci Please build toolchain macOS Platform

**Comment by QuietMisdreavus** [comment]
@swift-ci Please smoke test

**Comment by QuietMisdreavus** [comment]
@swift-ci Please test

File: lib/DriverTool/swift_symbolgraph_extract_main.cpp [hunk 1]
```diff
@@ -162,19 +162,18 @@ int swift_symbolgraph_extract_main(ArrayRef<const char *> Args,
...
-      !ParsedArgs.hasArg(OPT_skip_synthesized_members),
-      ParsedArgs.hasArg(OPT_v),
-      ParsedArgs.hasArg(OPT_skip_inherited_docs),
-      ParsedArgs.hasArg(OPT_include_spi_symbols),
-      /*IncludeClangDocs=*/false,
+  symbolgraphgen::SymbolGraphOptions Options;
+  Options.OutputDir = OutputDir;
+  Options.Target = Target;
+  Options.PrettyPrint = ParsedArgs.hasArg(OPT_pretty_print);
+  Options.EmitSynthesizedMembers = !ParsedArgs.hasArg(OPT_skip_synthesized_members);
```
Comment (franklinsch) [inline lib/DriverTool/swift_symbolgraph_extract_main.cpp:169]: Small nit on the naming here; the 'protocol implementations' one is exclusive (skip) and this one inclusive (emit).

**Review by franklinsch — APPROVED** [review]

**Comment by QuietMisdreavus** [comment]
@swift-ci Please build toolchain macOS Platform

**Comment by QuietMisdreavus** [comment]
@swift-ci Please build toolchain macOS Platform

**Pull request merged**